connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -206,16 +203,13 @@ public void cancel() {
     public void stop() {
         super.stop();
         stopRequestedLatch.countDown();
-        synchronized (this) {
-            if (finishedStart)
-                tryStop();
-            else
-                startedShutdownBeforeStartCompleted = true;
-        }
     }
 
-    private synchronized void tryStop() {
-        if (!stopped) {
+    // Note: This method is not thread-safe
```

Comment:
This is a fair point. TBH I don't think this method deserves to exist anymore at all and can just be folded into `close` where a bunch of similar blocks already exist. There's also no need to track `stopped` since `close` is only ever invoked once. I'll refactor and remove the comment entirely since everything should be more readable after that change even without commenting.

Reply:
Yeah, it's unclear whether to keep the `tryStop()` method to minimize deltas, or remove it to simplify things. +1 for simplifying things by moving the logic into `close()`.